### ISSUE TITLE
Forbid constants in `while`

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1006]
# Rule request

## Thesis

The only constant in `while` condition that makes sense is `True` for the "infinite" loop.

Bad:

```python
while 1: ...
while False: ...
while "infinity and beyond": ...
```

Good:

```python
while x: ...
while True: ...
```

## Reasoning

Consistency and fighting typos.

Interesting fact: for Python <=2.7 `while 1` was faster than `while True`. Luckily, it's not the case for Python 3.

**Comment by sobolevn** [comment]
Agree. We can limit nodes that `while` accepts to `NameConstant`, `Call`, `Name`, `Attribute`, `Subscript`. And also check that `NameConstant` is `True`, bot any other value.

**Comment by sobolevn** [comment]
Btw, it is also partially supported:

```python
ex.py

  2:7      WPS444 Found wrong keyword condition: False
  while False: ...
        ^

```
